From: SJ Park <sj@kernel.org>
Date: Sat, 8 Aug 2026 10:03:03 -0700
Subject: [PATCH] ==== vaddr: support probes ====

Test
====

I confirmed it can capture ~47 mb working set of masim in vaddr mode,
like below.

First, start masim [1] to access ~47 mb memory at a time, in background.

    $ ./masim/masim ./masim/configs/stairs-50mb.cfg --repeat 10 --quiet &

Note that the config says the working set is 50mb.  It is is 50 million
bytes, so ~47.68 MiB.

Start traditional access monitoring of masim's virtual address space
using damo [2].

    $ sudo ./damo/damo start $(pidof masim)

Confirm it can capture the 47 MiB working set.

    $ sudo ./damo report access
    heatmap: 0000000000000000000000000000000000000000000000036[...]40000000000000002448995444444434[...]7
    # min/max temperatures: -1,330,000,000, 170,010,000, column size: 15.447 MiB
    intervals: sample 5 ms aggr 100 ms (max access hz 200)
    0   addr 86.317 TiB   size 122.605 MiB access 0 hz   age 13.300 s
    1   addr 86.317 TiB   size 122.574 MiB access 0 hz   age 13.200 s
    2   addr 86.317 TiB   size 242.652 MiB access 0 hz   age 13.100 s
    3   addr 86.318 TiB   size 121.402 MiB access 0 hz   age 12.800 s
    4   addr 86.318 TiB   size 123.238 MiB access 0 hz   age 12.100 s
    5   addr 86.318 TiB   size 19.137 MiB  access 0 hz   age 3.600 s
    6   addr 127.484 TiB  size 123.578 MiB access 0 hz   age 13.100 s
    7   addr 127.484 TiB  size 123.160 MiB access 0 hz   age 12.700 s
    8   addr 127.484 TiB  size 39.430 MiB  access 0 hz   age 6.600 s
    9   addr 127.484 TiB  size 47.641 MiB  access 200 hz age 1.700 s
    10  addr 127.484 TiB  size 8.000 KiB   access 130 hz age 100 ms
    11  addr 127.484 TiB  size 121.219 MiB access 0 hz   age 5.400 s
    12  addr 127.484 TiB  size 28.988 MiB  access 0 hz   age 7.500 s
    13  addr 128.000 TiB  size 116.000 KiB access 0 hz   age 12.800 s
    14  addr 128.000 TiB  size 8.000 KiB   access 30 hz  age 0 ns
    15  addr 128.000 TiB  size 8.000 KiB   access 0 hz   age 12.400 s
    memory bw estimate: 9.306 GiB per second
    total size: 1.207 GiB
    record DAMON intervals: sample 5 ms, aggr 100 ms

Stop access monitoring and start probe-only mode access monitoring.

    $ sudo ./damo stop
    $ sudo ./damo start $(pidof masim) \
            --probe_prep set_pgidle --probe_filter allow pgidle_unset \
            --probe_weight 1

Confirm it can also capture the ~47 MiB working set.

    $ sudo ./damo report attrs
    heatmap: 0000000000000000000000000000000000000000000000028[...]43333333677777778876666666666667[...]9
    # min/max temperatures: -3,580,000,000, -27,510,778, column size: 15.447 MiB
    probe prep: set_pgidle, filter: allow pgidle_unset (weight: 1)
    intervals: sample 5 ms aggr 100 ms (max probe hits 20)
    #   addr         size         age           probe_hits
    0   86.317 TiB   123.578 MiB  34.700 s      0
    1   86.317 TiB   122.906 MiB  35.800 s      0
    2   86.317 TiB   123.449 MiB  35.400 s      0
    3   86.318 TiB   122.953 MiB  35.200 s      0
    4   86.318 TiB   123.012 MiB  34.800 s      0
    5   86.318 TiB   122.254 MiB  33.800 s      0
    6   86.318 TiB   13.461 MiB   900 ms        0
    7   127.484 TiB  122.457 MiB  23.500 s      0
    8   127.484 TiB  115.977 MiB  7.300 s       0
    9   127.484 TiB  8.000 KiB    100 ms        10
    10  127.484 TiB  47.695 MiB   3.300 s       20
    11  127.484 TiB  122.801 MiB  8.600 s       0
    12  127.484 TiB  75.086 MiB   10 s          0
    13  128.000 TiB  116.000 KiB  35.100 s      0
    14  128.000 TiB  8.000 KiB    100 ms        14
    15  128.000 TiB  8.000 KiB    34.900 s      0
    memory bw estimate: 0 B per second
    total size: 1.207 GiB
    record DAMON intervals: sample 5 ms, aggr 100 ms

[1] https://github.com/sjp38/masim
[2] https://github.com/damonitor/damo

Signed-off-by: SJ Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_i6z435or | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_i6z435or

diff --git a/hkml_cv_bogus/hkml_cv_bogus_i6z435or b/hkml_cv_bogus/hkml_cv_bogus_i6z435or
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

